### PR TITLE
doc: usermode: fix syscall_handler.h location in docs

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Trivial

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: doc/reference/usermode/syscalls.rst [hunk 1]
```diff
@@ -241,7 +241,7 @@ passed in.  This includes:
 * Any other arguments that have a limited range of valid values.
 
 Verification functions involve a great deal of boilerplate code which has been
-made simpler by some macros in ``kernel/include/syscall_handlers.h``.
+made simpler by some macros in ``kernel/include/syscall_handler.h``.
```
Comment (andrewboie) [inline doc/reference/usermode/syscalls.rst:244]: I don't recall when this got moved but it's in `include/syscall_handler.h`, it's no longer in the private kernel include directory.

Comment (mbolivar-nordic) [reply doc/reference/usermode/syscalls.rst]: Good catch. Fixed and replaced with `:zephyr_file:` so the doc build will fail if this goes stale again.

**Comment by ioannisg** [comment]
@mbolivar-nordic maybe rebase this one?

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
> @mbolivar-nordic maybe rebase this one?

Done, thanks.

**Comment by ioannisg** [comment]
Still appears with conflicts, to me.

**Comment by de-nordic** [comment]
@mbolivar-nordic Something went wrong with the rebase, I have altered the line in commit 9dc00e45ce01e80fec188161dc56a8278c6c02e4 and it should look different.

**Comment by mbolivar-nordic** [comment]
Ah, I see you fixed the path along with other things. I'll still rebase this as I think using `:zephyr_file:` is a bit cleaner since it's been reviewed. Rebased properly this time.

**Pull request merged**